### DISCUSSION TITLE
v1/en/docs/getting-started/installation/git-submodule/

### DISCUSSION THREAD

**Discussion by giscus** [razonyang/hugo-theme-bootstrap#654] [General]
# v1/en/docs/getting-started/installation/git-submodule/

This article refers to the traditional installation method of cloning the theme locally as Git submodule. While this installation method is still widely used, we recommend to install the theme as Hugo module.
Create a New Site from Scratch# 1$ git clone https://github.com/razonyang/hugo-theme-bootstrap-skeleton myblog 2$ cd myblog 3$ git submodule add https://github.com/razonyang/hugo-theme-bootstrap themes/hugo-theme-bootstrap 4$ sed -i "s/theme:.*/theme: hugo-theme-bootstrap/g" config/_default/config.yaml 5$ rm go.mod go.sum 6$ hugo mod npm pack 7$ npm install 8$ hugo server  If you’re using Windows, use xcopy .

https://hbs.razonyang.com/v1/en/docs/getting-started/installation/git-submodule/

**Comment by kuaileniu** [comment]
这个地方的功能需要写服务器端的代码吗？

**Reply by razonyang** [reply]
😕 不需要的
